### PR TITLE
Feature/sidebar v2

### PR DESCRIPTION
**Pull request by Akswii**
Added new component sidebar.

### CODE REVIEW

File: src/px-script/sidebar/index.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        document.addEventListener("click", e => {
+            if (!e.target.closest(".sidebar") && this.sidebarOpen) {
+                this.closeSidebar();
+                this.hideItems();
+            } else if (!e.target.closest(".submenu") && !this.submenuClosed) {
+                this.submenuCloseAll();
+            }
+        });
+
+        if (this.listItems > 5 || this.childCount > 0) {
```
Comment (erikhallaraaker) [inline src/px-script/sidebar/index.js]: No need to check specifically if it is bigger than `0`

```suggestion
        if (this.listItems > 5 || this.childCount) {
```

File: src/px-script/sidebar/index.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            const menu = document.createElement("a");
+
+            menu.setAttribute("href", "#");
+            this.hideItems();
+            menu.classList.add("sidebar-openbtn");
+            menu.innerHTML += "<i class='material-icons'>menu</i>";
```
Comment (erikhallaraaker) [inline src/px-script/sidebar/index.js]: Unless you plan on keeping the content of the `innerHTML` (which on this line is nothing), you dont need the `+`

```suggestion
            menu.innerHTML = "<i class='material-icons'>menu</i>";
```

File: src/px-script/sidebar/index.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            } else if (!e.target.closest(".submenu") && !this.submenuClosed) {
+                this.submenuCloseAll();
+            }
+        });
+
+        if (this.listItems > 5 || this.childCount > 0) {
+            const menu = document.createElement("a");
+
+            menu.setAttribute("href", "#");
+            this.hideItems();
```
Comment (erikhallaraaker) [inline src/px-script/sidebar/index.js]: I would move this line either above or below the `menu`-"block" to keep things easier to read.

File: src/px-script/sidebar/index.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        }
+    }
+
+    onResize () {
+        this.closeSidebar();
+        this.hideItems();
+        this.submenuCloseAll();
+    }
+
+    openSidebar () {
```
Comment (erikhallaraaker) [inline src/px-script/sidebar/index.js]: I would rename this method to just `open`, it is a method only used by the `Sidebar` class.
```javascript
const sidebar = new Sidebar();
sidebar.open();
```
Looks better than
```javascript
const sidebar = new Sidebar();
sidebar.openSidebar();
```

File: src/px-script/sidebar/index.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        this.sidebarOpen = true;
+        this._el.classList.add("sidebar-open");
+        this.resizeEventMenuOpen = this.onResize.bind(this);
+        window.addEventListener("resize", this.resizeEventMenuOpen, { passive: true });
+    }
+
+    closeSidebar () {
```
Comment (erikhallaraaker) [inline src/px-script/sidebar/index.js]: Same as for the comment above on the `openSidebar` method.

File: src/App/Documentation/components/Sidebar/index.js [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+        icon: "account_balance",
+        subItems: ["Email", "Information", "Other things"]
+    }
+];
+
+const StandardSidebar = () => (
+    <>
+        <h2 id="sidebar-standard">Standard sidebar</h2>
+        <p>
+            Add class <code className="token property">.sidebar</code> and <code className="token property">.sidebar-(desired_size)-vertical-wide</code> to a nav containing an <PrismCode className="language-html">{"<ul>"}</PrismCode> to get a standard sidebar.
```
[the remark this answers is not in the document]
Comment (Akswii) [reply src/App/Documentation/components/Sidebar/index.js]: I don't currently have  `<Attribute />` and `<Property />` available. Using brackets to show where you can use desired_size makes it so it's read as a variable.

[the remark this answers is not in the document]
Comment (Akswii) [reply src/App/Documentation/components/Sidebar/index.js]: I don't currently have  `<Attribute />` and `<Property />` available. Using brackets to show where you can use desired_size makes it so it's read as a variable.

[the remark this answers is not in the document]
Comment (Akswii) [reply src/App/Documentation/components/Sidebar/index.js]: I don't currently have  `<Attribute />` and `<Property />` available. Using brackets to show where you can use desired_size makes it so it's read as a variable.

**Pull request merged**